inmemory_repository.go

```diff
@@ -27,7 +27,7 @@ func (r *InMemoryRepository) Save(object DomainObject) error {
 	return nil
 }
 
-func (r *InMemoryRepository) Load(objectID string, object DomainObject) error {
+func (r *InMemoryRepository[T]) Load(objectID string, object T) error {
```

Comment:
```suggestion
func (r *InMemoryRepository[T]) Load(objectID string) (T, error) {
```

Mais peut-être que ça fait trop de changements ?

Reply:
J'ai tenté de faire ça ! Ça faisait effectivement pas mal de changement mais j'ai surtout eu un souci dans la création de l'objet dans le load. Je n'ai plus les détails exacts en tête mais je pense que c'était une question d'interface et de pointer receviez. Un truc du style que l'objet créé n'avait pas d'EventStream parce qu'EventStream est une interface pour laquelle on passe une implémentation avec un pointer receiver. J'ai cherché pendant une petite heure puis j'ai revert. Note que je pense que ce serait envisageable d'y arriver par contre. Il doit bien y avoir un moyen. À voir si on veut le faire dans cette MR ou pas.